src/cdk/a11y/focus-monitor/focus-monitor.ts

```diff
@@ -134,7 +137,15 @@ export class FocusMonitor implements OnDestroy {
     this._windowFocusTimeoutId = setTimeout(() => this._windowFocused = false);
   }
 
-  constructor(private _ngZone: NgZone, private _platform: Platform) {}
+   /** Used to reference correct document/window */
+   protected _document?: Document;
+
+  constructor(private _ngZone: NgZone,
+              private _platform: Platform,
+              /** @breaking-change 11.0.0 make document required */
+              @Optional() @Inject(DOCUMENT) document?: any) {
```

Comment:
you can combine the param and field by doing:
`@Optional() @Inject(DOCUMENT) protected _document?: any`

Reply:
No, we actually want it this way. The token has to be `any` when injected due to the decorator (SSR complications), but we still want typing for `Document` in the rest of the file.